registrarc/registrarc_test.go

```diff
@@ -123,3 +123,104 @@ func TestDispatcherHostsService(t *testing.T) {
 	close(stopChan)
 	ds.ListenAndServe(stopChan, make(chan struct{}))
 }
+
+func TestRegistrarcListenAndServe(t *testing.T) {
+	//cover purposes only
+	cfg := config.NewDefaultCGRConfig()
+	cfg.RegistrarCCfg().Dispatcher.Enabled = true
+	cfg.RegistrarCCfg().RPC.Enabled = true
+	regStSrv := NewRegistrarCService(cfg, nil)
+	stopChan := make(chan struct{}, 1)
+	rldChan := make(chan struct{}, 1)
+	rldChan <- struct{}{}
+	go func() {
+		time.Sleep(10 * time.Millisecond)
+		close(stopChan)
+	}()
+	regStSrv.ListenAndServe(stopChan, rldChan)
+	regStSrv.Shutdown()
+}
+
+func TestRegistrarcregisterRPCHostsErr(t *testing.T) {
+	cfg := config.NewDefaultCGRConfig()
+	cfg.RegistrarCCfg().RPC.RegistrarSConns = []string{"testConnID"}
+	cfg.RegistrarCCfg().RPC.Hosts = map[string][]*config.RemoteHost{
+		utils.MetaDefault: {
+			{
+				ID:          "",
+				Address:     "",
+				Transport:   "",
+				Synchronous: false,
+				TLS:         false,
+			},
+		},
+	}
+	regStSrv := NewRegistrarCService(cfg, nil)
+	regStSrv.registerRPCHosts()
+}
+
+func TestRegisterRPCHosts(t *testing.T) {
+	cfg := config.NewDefaultCGRConfig()
+	cfg.RegistrarCCfg().RPC.RegistrarSConns = []string{"errCon1"}
+	cfg.RegistrarCCfg().RPC.Hosts = map[string][]*config.RemoteHost{
+		"testHostKey": {},
+	}
+	cfg.RPCConns()["errCon1"] = &config.RPCConn{
+		Strategy: utils.MetaFirst,
+		PoolSize: 1,
+		Conns: []*config.RemoteHost{
+			{
+				ID:          "errCon1",
+				Address:     "127.0.0.1:9999",
+				Transport:   "*json",
+				Synchronous: true,
+			},
+		},
+	}
+	regist := &RegistrarCService{
+		cfg:     cfg,
+		connMgr: engine.NewConnManager(cfg, map[string]chan rpcclient.ClientConnector{}),
+	}
+	registCmp := &RegistrarCService{
+		cfg:     cfg,
+		connMgr: engine.NewConnManager(cfg, map[string]chan rpcclient.ClientConnector{}),
+	}
+	regist.registerRPCHosts()
+	if !reflect.DeepEqual(regist, registCmp) {
+		t.Errorf("Expected: %+v ,received: %+v", registCmp, regist)
+	}
+}
+
+func TestRegistrarcListenAndServedTmCDispatcher(t *testing.T) {
```

Comment:
Please rename without Test function since there is no failing test here. Same in other places where we do not test.